components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts

```diff
@@ -142,6 +147,34 @@ export default function shieldsPanelReducer (
         })
       break
     }
+
+    case shieldsPanelTypes.SPEEDREADER_TOGGLED: {
+      console.log("reducer:: shieldsPanelTypes.SPEEDREADER_TOGGLE")
+
+      const tabData = shieldsPanelState.getActiveTabData(state)
+      if (!tabData) {
+        console.error('Active tab not found')
+        break
+      }
+
+      setEnableSpeedReader(tabData.origin, toggleShieldsValue(action.setting))
+        .then(() => {
+          requestShieldPanelData(shieldsPanelState.getActiveTabId(state))
+          reloadTab(tabData.id, true).catch(() => {
+            console.error('Tab reload was not successful')
+        })
+       })
+       .catch(() => {
+         console.error('Could not enable Speedreader setting')
+       })
+       break
+
+       //const tabId: number = shieldsPanelState.getActiveTabId(state)
```

Comment:
nit: run `npm run lint` from within `src/brave` to catch some linting erorrs in these files.